types/int.go

```diff
@@ -525,3 +526,10 @@ func (i *Uint) UnmarshalJSON(bz []byte) error {
 	}
 	return unmarshalJSON(i.i, bz)
 }
+
+//__________________________________________________________________________
+
+// intended to be used with require/assert:  require.True(IntEq(...))
+func IntEq(t *testing.T, exp, got Int) (*testing.T, bool, string, string, string) {
```

Comment:
For a separate PR / issue, but I don't think functions like this should be exported. At most, they could be in an internal package. However I think using such a function as a design pattern is bad. (Unnecessarily crowds the public API surface)

Reply:
Yeah cool - maybe it could be exported as a special testing-helpers package, I DO find these helper functions to be super useful / make writing tests less painful (for which we should be striving to do)

Reply:
- maybe a cleaner pattern would actually just be having pre-fab assert/require functions for some of the sdk primitives - therefor, we're not passing the outs from one function into the in's of the next function.

Reply:
openned issue https://github.com/cosmos/cosmos-sdk/issues/2494